src/battle.ts

```diff
@@ -1187,10 +1196,12 @@ class Battle {
 	}
 	switchSides() {
 		this.setSidesSwitched(!this.sidesSwitched);
+		this.scene.updateSidebars();
 		this.resetToCurrentTurn();
 	}
 	setSidesSwitched(sidesSwitched: boolean) {
 		this.sidesSwitched = sidesSwitched;
+		if (this.mySide === this.p3 || this.mySide === this.p4) return;
 		if (this.sidesSwitched) {
 			this.nearSide = this.mySide = this.p2;
```

Comment:
Wait, shouldn't `this.mySide` never change? I thought that was the point of separating `nearSide` and `mySide`?

Reply:
I'm pretty sure anything that uses `mySide` is exclusive to actual players and that all the code used for spectators only ever uses `nearSide`, so assuming the code that sets `mySide` when a player joins actually does work for p2, then this doesn't need to set `mySide`, but only because it doesn't matter.

Reply:
I'm saying that it's bad for this to set `mySide`, because it would cause this function to break when a player joins. Because of that, this should be taken out.

Reply:
In fact, I think it would make a lot of sense for `.mySide` to be `null` if the player isn't in a battle. I think that would make it easier to reason about.